markdown/projects/mr-robot/major-code/arduino.rst

```diff
@@ -0,0 +1,113 @@
+Arduino
+=======
+
+.. code:: C
+
+    void LpwmCb(const std_msgs::Int32& pwm){
+
+        if(pwm.data == 0) { ledcWrite(ledChannel1, 0); ledcWrite(ledChannel2, 0); }
+        if(pwm.data > 0)  { ledcWrite(ledChannel1, abs(pwm.data)); ledcWrite(ledChannel2, 0); }
+        if(pwm.data < 0)  { ledcWrite(ledChannel1, 0); ledcWrite(ledChannel2, abs(pwm.data));}
+    }
+
+This code defines two callback functions, LpwmCb and RpwmCb, which take a std_msgs::Int32 object as input. The functions control the speed of four motors by writing PWM signals using the ledcWrite function.
+
+- The if statements check the value of the input pwm.data and determine which PWM signal to write to each motor. 
+- If pwm.data is zero, then both PWM signals for the corresponding motor are set to zero, which stops the motor. 
+- If pwm.data is positive, then the first PWM signal is set to the absolute value of pwm.data, which sets the motor to move forward. 
+- If pwm.data is negative, then the second PWM signal is set to the absolute value of pwm.data, which sets the motor to move backward.
+
+.. code:: C
+
+    void setup() {
+
+      nh.initNode();
+    
+      nh.advertise(left_enc_pub);
+      nh.advertise(right_enc_pub);
+      nh.advertise(left_enc_error);
+      nh.advertise(right_enc_error);
+
+      nh.subscribe(Lpwm_sub);
+      nh.subscribe(Rpwm_sub);
+
+      ledcSetup(ledChannel1, freq, resolution);
+      ledcSetup(ledChannel2, freq, resolution);
+      ledcSetup(ledChannel3, freq, resolution);
+      ledcSetup(ledChannel4, freq, resolution);
+
+      pinMode(encoderLPin1, INPUT_PULLUP); 
+      pinMode(encoderLPin2, INPUT_PULLUP);
+      pinMode(encoderRPin1, INPUT_PULLUP); 
+      pinMode(encoderRPin2, INPUT_PULLUP);
+
+      pinMode(standby, OUTPUT);
+      digitalWrite(standby, HIGH);
+
+      ledcAttachPin(LForward,      ledChannel1);
+      ledcAttachPin(LBackward,     ledChannel2);
+      ledcAttachPin(RForward,      ledChannel3);
+      ledcAttachPin(RBackward,     ledChannel4);
+
+      lastStateL = digitalRead(encoderLPin1);
+      lastStateR = digitalRead(encoderRPin1);
+
+      attachInterrupt(digitalPinToInterrupt(encoderLPin1), updateEncoder_L, CHANGE);
+      attachInterrupt(digitalPinToInterrupt(encoderLPin2), updateEncoder_L, CHANGE);
+    
+      attachInterrupt(digitalPinToInterrupt(encoderRPin1), updateEncoder_R, CHANGE);
+      attachInterrupt(digitalPinToInterrupt(encoderRPin2), updateEncoder_R, CHANGE);
+
+
+    }
+
+This code snippet is used to set up the various peripherals and settings needed for the robot to operate, including the ROS node, PWM signals for the motors, encoder signals for the wheels, and interrupt service routines for updating the encoder counts. 
```

Comment:
Peripherals is not the right word to be used to address the components/parts here